src/main.rs

```diff
@@ -29,55 +28,39 @@ use error::{Error, Result};
 
 static NOTFOUND: &[u8] = b"Not Found";
 
-#[tokio::main]
+#[actix_web::main]
 async fn main() -> Result<()> {
-    match run().await {
-        Ok(_) => Ok(()),
-        Err(e) => {
-            println!("Error occured: {}", e);
-            std::process::exit(1);
-        }
-    }
-}
-
-async fn run() -> Result<()> {
     pretty_env_logger::init();
-    // Get a context to work with the TPM
     let mut ctx = tpm::get_tpm2_ctx()?;
-
-    // queue up future events
-    let server = runWebServer();
-    let revoker = runRevocationService();
-
-    // run future events
-    try_join!(server, revoker)?;
-    Ok(())
-}
-
-async fn runWebServer() -> Result<()> {
     let cloudagent_ip =
         config_get("/etc/keylime.conf", "cloud_agent", "cloudagent_ip")?;
     let cloudagent_port =
         config_get("/etc/keylime.conf", "cloud_agent", "cloudagent_port")?;
-    let endpoint = format!("{}:{}", cloudagent_ip, cloudagent_port);
-
     info!("Starting server...");
-
-    let addr = (endpoint).parse().expect("Cannot parse IP & Port");
-
-    let service = make_service_fn(|_| async {
-        Ok::<_, Error>(service_fn(handlers::response_function))
-    });
-    let server = Server::bind(&addr).serve(service);
-
-    info!("Listening on http://{}", addr);
-
-    server.await?;
+    let actix_server = HttpServer::new(move || {
```

Comment:
The mounting of routes and the creation of the server directly seems like it would make sense in another file and used here. However I think this is good as is and can be changed later _if_ we decide to do so.

Reply:
I will have a play with that , something like `routes.rs` , it may be tricky though because of actix being heavily future based. I had a good look on github searches , but not many folks were separating the routes out of main. I do agree though, its a cleaner way of doing it.